src/mlpack/methods/reinforcement_learning/environment/continuous_mountain_car_V2.hpp

```diff
@@ -0,0 +1,224 @@
+   * discrete value, continuous mountain car has continous action space
+   * value.
+   */
+  struct Action
+  {
+    double action[1];
+    // Storing degree of freedom
+    const int size = 1;
+  };
+
+  /**
+   * Construct a Continuous Mountain Car V2 instance using the given constant.
+   *
+   * @param positionMin Minimum legal position.
+   * @param positionMax Maximum legal position.
+   * @param positionGoal Final target position.
+   * @param velocityMin Minimum legal velocity.
+   * @param velocityMax Maximum legal velocity.
+   * @param velocityGoal Final target velocity.
+   * @param power Power generated by car.
+   */
+  ContinuousMountainCarV2(const double positionMin = -1.2,
+                        const double positionMax = 0.6,
+                        const double positionGoal = 0.45,
+                        const double velocityMin = -0.07,
+                        const double velocityMax = 0.07,
+                        const double velocityGoal = 0.06,
+                        const double power = 0.0015) :
+      positionMin(positionMin),
+      positionMax(positionMax),
+      positionGoal(positionGoal),
+      velocityMin(velocityMin),
+      velocityMax(velocityMax),
+      velocityGoal(velocityGoal),
+      power(power)
+  { /* Nothing to do here */ }
+
+  /**
+   * Dynamics of Continuous Mountain CarV2. Get reward and next state based
+   * on current state and current action.
+   *
+   * @param state The current state.
+   * @param action The current action.
+   * @param nextState The next state.
+   * @return reward, it's always -1.0.
+   */
+  double Sample(const State& state,
+                const Action& action,
+                State& nextState) const
+  {
+    // Calculate acceleration.
+    double force = std::min(std::max(action.action[0], -1.0), 1.0);
+
+    // Update states.
+    nextState.Velocity() = state.Velocity() + force * power - 0.0025 *
+        std::cos(3 * state.Position());
+    nextState.Velocity() = std::min(
+        std::max(nextState.Velocity(), velocityMin), velocityMax);
+    nextState.Position() = state.Position() + nextState.Velocity();
+    nextState.Position() = std::min(
+        std::max(nextState.Position(), positionMin), positionMax);
+    if (nextState.Position() == positionMin && nextState.Velocity() < 0)
+      nextState.Velocity() = 0.0;
+
+    // Calculate reward
+    double reward = 0.0;
+    // If it is a terminal state, add a reward of 100.0
+    if (IsTerminal(nextState))
+      reward = 100.0;
+    reward -= std::pow(action.action[0], 2) * 0.1;
+    return reward;
+  }
+
+  /**
+   * Dynamics of Continuous Mountain Car V2. Get reward based on current state and current
+   * action.
+   *
+   * @param state The current state.
+   * @param action The current action.
+   * @return reward, it's always -1.0.
+   */
+  double Sample(const State& state, const Action& action) const
+  {
+    ContinuousMountainCarV2 ob = ContinuousMountainCarV2(positionMin,
```

Comment:
This will call the ``Sample``function from the ``ContinuousMountainCarV2`` class, but I was thinking about using the ``Sample`` function from the ``ContinuousMountainCar`` class since it's the same.